addons/web/static/src/webclient/user_menu/user_menu.xml

```diff
@@ -17,6 +17,7 @@
                     <t t-set-slot="content">
                         <t t-foreach="getElements()" t-as="element" t-key="element_index">
                             <t t-if="!element.hide">
+                                <t t-if="element.type === 'component'" t-component="element.contentComponent"/>
```

Comment:
Add a simple test in `web` with a simple example component to cover this new option.